File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for n in s:
            count = 1
            while n + 1 in s:
                count += 1
                n = n+1
            
            res = max(count, res)

        return res

File: Data Structures & Algorithms/combination-target-sum-ii/submission-4.cpp
class Solution {
public:
    vector<vector<int>> combinationSum2(vector<int>& candidates, int target) {
        vector<vector<int>> res;
        vector<int> cur;
        sort(candidates.begin(), candidates.end());
        dfs(0, res, 0, cur, candidates, target);
        return res;

    }

private:
    void dfs(int idx, vector<vector<int>>& res, int sum, vector<int> cur, vector<int>& nums, int target){
        if(sum == target){
            res.push_back(cur);
            return;
        }
        for(int i =idx; i<nums.size(); i++){
            if(i>idx && nums[i] == nums[i -1]){
                continue;
            }
            if(sum + nums[i] > target){
                break;
            }
            cur.push_back(nums[i]);
            dfs(i+1, res, sum+nums[i],cur, nums,target);
            cur.pop_back();
        }
    }
};
